app/routes.py

```diff
@@ -1,2 +1,109 @@
+        else:
+            tasks = Task.query.all()
+        for task in tasks:
+            task_list.append({
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at is None else True
+            })
+        return jsonify(task_list)
+
+    elif request.method =="POST":
+        request_body = request.get_json()
+        
+        if 'title' not in request_body or 'description' not in request_body or 'completed_at' not in request_body:
+            return  {"details": "Invalid data"},400
+
+        new_task = Task(title=request_body["title"], 
+            description=request_body["description"], 
+            completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        
+        response_body = {'task': {
+            "id": new_task.id, 
+            "title": new_task.title, 
+            "description": new_task.description, 
+            "is_complete": False if new_task.completed_at is None else True}}
+
+
+        return response_body, 201
+    
+@task_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if request.method =="GET":
+        if task is None:
+            return make_response("", 404)
+        return make_response({
+            "task":{
+            "id": task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at is None else True
+                }
+        }, 200)
+
+    elif request.method == "PUT":
+        if task is None:
+            return make_response("", 404)
+        form_data = request.get_json()
+
+        task.title=form_data["title"]
+        task.description = form_data["description"]
+        # new_task = jsonify(task)
+        response_body={"task": {"id": task.id, "title": task.title, "description": task.description, "is_complete": False if task.completed_at is None else True}}
+
+        db.session.commit()
+        return response_body, 200
+
+    elif request.method =="DELETE":
+        if task is None:
+            return make_response("", 404)
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details":f'Task {task.id} "{task.title}" successfully deleted'}, 200)
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task=Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+    
+    db.session.commit()
+    task.completed_at=True
+    response_body={"task": {"id": task.id, "title": task.title, "description": task.description, "is_complete": False if task.completed_at is None else True}}
+    r = requests.post(url="/<chat.postMessage>", channel="task-notifications", text=f'Someone completed the task {task.title}')
```

Comment:
for the slack information I would create variables to hold the information then pass them into the `requests.post` something like this 
```suggestion
   headers = {
        "Authorization": f"Bearer {SLACK_BOT_TOKEN}",
    }
    if task.completed_at:
        data = {
            "channel": "general",
            "text": f"Task {task.title} has been marked complete",
        }
    else: 
        data = {
            "channel": "general",
            "text": f"Task {task.title} has been marked incomplete",
        }
    r = requests.post(SLACK_API_URL, headers=headers, data=data )
```